src/platform/windows/display.h

```diff
@@ -338,7 +357,53 @@ namespace platf::dxgi {
     texture2d_t old_surface_delayed_destruction;
     std::chrono::steady_clock::time_point old_surface_timestamp;
     std::variant<std::monostate, texture2d_t, std::shared_ptr<platf::img_t>> last_frame_variant;
+  };
 
-    std::atomic<uint32_t> next_image_id;
+  class uwp_capture_t {
+    winrt::Windows::Graphics::DirectX::Direct3D11::IDirect3DDevice uwp_device { nullptr };
+    winrt::Windows::Graphics::Capture::GraphicsCaptureItem item { nullptr };
+    winrt::Windows::Graphics::Capture::Direct3D11CaptureFramePool frame_pool { nullptr };
+    winrt::Windows::Graphics::Capture::GraphicsCaptureSession capture_session { nullptr };
+    winrt::Windows::Graphics::Capture::Direct3D11CaptureFrame capture_frame { nullptr };
+    CRITICAL_SECTION frame_lock;
```

Comment:
We should prefer `SRWLOCK` which is slightly faster than critical sections, even if we don't use the support for multiple concurrent readers.